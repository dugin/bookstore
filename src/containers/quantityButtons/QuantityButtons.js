import React, {Component} from 'react';


class QuantityButtons extends Component {

    constructor(props) {
        super(props);

        this.state = {quantity: 1};
    }

    onAdd = () => {
        this.setState(state => state.quantity++);
        this.props.onAddOrRemove(this.state.quantity)
    };

    onRemove = () => {
        this.setState(state => state.quantity > 1 ? state.quantity-- : 1);

        this.props.onAddOrRemove(this.state.quantity)
    };

    render() {
        return (
            <div className="Book__actions__quantityContainer">
                <button onClick={this.onRemove} className="Book__actions__quantityContainer__minus"><i
                    className="material-icons">remove</i>
                </button>

                <span className="Book__actions__quantityContainer__quantity"> {this.state.quantity} </span>

                <button onClick={this.onAdd} className="Book__actions__quantityContainer__plus "><i
                    className="material-icons">add</i>
                </button>
            </div>
        )
    }
}

export default QuantityButtons;