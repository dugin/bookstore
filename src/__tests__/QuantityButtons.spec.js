import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import QuantityButtons from "../containers/quantityButtons/QuantityButtons";

describe('QuantityButtons', () => {

    const book = shallow(<QuantityButtons onAddOrRemove={jest.fn()}/>);

    const plusBtn = book.find('.Book__actions__quantityContainer__plus');
    const minusBtn = book.find('.Book__actions__quantityContainer__minus');

    beforeEach(() => {
        book.setState({quantity: 1});
    });

    it('renders properly', () => {
        expect(toJson(book)).toMatchSnapshot();
    });


    it('adds one on plus button click ', () => {
        expect(plusBtn.length).toEqual(1);
        plusBtn.simulate('click');

        expect(book.state().quantity).toEqual(2);
    });

    it('do not subtract on minus button if quantity == 1', () => {
        expect(plusBtn.length).toEqual(1);
        expect(minusBtn.length).toEqual(1);

        minusBtn.simulate('click');

        expect(book.state().quantity).toEqual(1);
    });

    it('subtract one on minus button click if quantity > 1', () => {
        book.setState({quantity: 3});

        expect(minusBtn.length).toEqual(1);
        minusBtn.simulate('click');

        expect(book.state().quantity).toEqual(2);
    });


});
