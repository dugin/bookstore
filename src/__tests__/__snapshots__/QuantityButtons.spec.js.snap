// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityButtons renders properly 1`] = `
<div
  className="Book__actions__quantityContainer"
>
  <button
    className="Book__actions__quantityContainer__minus"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      remove
    </i>
  </button>
  <span
    className="Book__actions__quantityContainer__quantity"
  >
     
    1
     
  </span>
  <button
    className="Book__actions__quantityContainer__plus "
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      add
    </i>
  </button>
</div>
`;
